pkg/plugins/runtime/k8s/controllers/gateway_instance_controller.go

```diff
@@ -31,9 +35,16 @@ import (
 	util_k8s "github.com/kumahq/kuma/pkg/plugins/runtime/k8s/util"
 )
 
+const (
+	// FailedToCreateService is added to an event when
+	// the Service for a GatewayInstance can't be created.
+	FailedToCreateService = "FailedToCreateService"
```

Comment:
I'm not sure this is the right event name to publish. As used, it doesn't indicate there was a failure (we didn't attempt to create a service and have that attempt fail), it indicates that we should not be creating a service and so we didn't, which is OK and normal.